### ISSUE TITLE
Implementing and using new syntactic features breaks linting

### ISSUE THREAD

**Issue by weswigham** [microsoft/TypeScript#33479]
Because the new linter, `eslint` can't be built against the nightly TS parser and can't understand any newly introduced syntax. Found over at #33290. To handle this, we'd need to build against the nightly-available fork of our parser, or whatever's locally available (so when you're working on a new syntactic feature you don't have broken lints), I think.

**Comment by weswigham** [comment]
This is actually a big problem, since it blocks [tarballs](https://typescript.visualstudio.com/TypeScript/_build/results?buildId=44398) of in-progress features, too... For now, I'll go disable linting on `tarball` to unblock them, but we really need a plan here.

**Comment by uniqueiniquity** [comment]
As discussed offline, there's no reason it can't use the nightly TS parser and in fact, it should.

**Comment by weswigham** [comment]
I'm gunna close this - I think this was mostly just different from before because I skipped the LKG phase of feature rollout in #33290 and the tarball build didn't initially fail (pre-eslint) to warn me about it. I've fixed up the build to fail correctly [as here](https://typescript.visualstudio.com/TypeScript/_build/results?buildId=44408). Now I'm just fixing up the order of operations in the PR so it can LKG, and I think I *should* be able to reenable linting then.
